plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -265,10 +349,83 @@ private List<Object> getRecord(StructLike structLike)
             columns.add(toBigintArrayBlock(structLike.get(columnNameToPosition.get(SPLIT_OFFSETS_COLUMN_NAME), List.class)));
             columns.add(toIntegerArrayBlock(structLike.get(columnNameToPosition.get(EQUALITY_IDS_COLUMN_NAME), List.class)));
+            columns.add(structLike.get(columnNameToPosition.get(SORT_ORDER_ID_COLUMN_NAME), Integer.class));
+
+            ReadableMetricsStruct readableMetrics = structLike.get(columnNameToPosition.get(READABLE_METRICS_COLUMN_NAME), ReadableMetricsStruct.class);
+            columns.add(toJson(readableMetrics));
+
             checkArgument(columns.size() == types.size(), "Expected %s types in row, but got %s values", types.size(), columns.size());
             return columns;
         }
 
+        private String toJson(ReadableMetricsStruct readableMetrics)
+        {
+            StringWriter writer = new StringWriter();
+            try {
+                JsonGenerator generator = JSON_FACTORY.createGenerator(writer);
+                generator.writeStartObject();
+
+                for (int i = 0; i < readableMetrics.size(); i++) {
+                    Types.NestedField field = primitiveFields.get(i);
+                    generator.writeFieldName(field.name());
+
+                    generator.writeStartObject();
+                    ReadableColMetricsStruct columnMetrics = readableMetrics.get(i, ReadableColMetricsStruct.class);
+
+                    generator.writeFieldName("column_size");
+                    Long columnSize = columnMetrics.get(0, Long.class);
+                    if (columnSize == null) {
+                        generator.writeNull();
+                    }
+                    else {
+                        generator.writeNumber(columnSize);
+                    }
+
+                    generator.writeFieldName("value_count");
+                    Long valueCount = columnMetrics.get(1, Long.class);
+                    if (valueCount == null) {
+                        generator.writeNull();
+                    }
+                    else {
+                        generator.writeNumber(valueCount);
+                    }
+
+                    generator.writeFieldName("null_value_count");
+                    Long nullValueCount = columnMetrics.get(2, Long.class);
+                    if (nullValueCount == null) {
+                        generator.writeNull();
+                    }
+                    else {
+                        generator.writeNumber(nullValueCount);
+                    }
+
+                    generator.writeFieldName("nan_value_count");
+                    Long nanValueCount = columnMetrics.get(3, Long.class);
+                    if (nanValueCount == null) {
+                        generator.writeNull();
+                    }
+                    else {
+                        generator.writeNumber(nanValueCount);
+                    }
+
+                    generator.writeFieldName("lower_bound");
```

Comment:
this could potentially be sensitive information which might be guarded by access control rules